SparseGrids/tsgCoreOneDimensional.cpp

```diff
@@ -392,6 +393,72 @@ const char* OneDimensionalMeta::getHumanString(TypeOneDRule rule){
     }
 }
 
+// Generate the n roots of the n-th degree orthogonal polynomial.
+// ref_points and ref_points are used in the computation of the integrals that form the elements of the Jacobi matrix.
+double poly_eval(int nroots, std::vector<double> roots, double x) {
+    double eval_value = 1.0;
+    for (int i=0; i<nroots; i++) {
+        eval_value *= (x - roots[i]);
+    }
+    return(eval_value);
+}
+std::vector<std::vector<double>> OneDimensionalOrthPolynomials::getRootCache(
+        int n,
+        std::function<double(double)> &weight_fn,
+        const std::vector<double> &ref_weights,
+        const std::vector<double> &ref_points) {
+
+    // Initialize and compute the integral weights corresponding to weight_fn.
+    std::vector<std::vector<double>> root_cache(n);
+    std::vector<double> roots;
+    #ifdef Tasmanian_ENABLE_BLAS
+    if (ref_points.size() != ref_weights.size()) {
```

Comment:
This is a check for the internals, we can just use `assert()`:

```
assert(ref_points.size() == ref_weights.size());
```
The purpose of the exceptions is that the user receives feedback on what went wrong, we don't need to be this verbose in our own internal error checking.

Reply:
Okay. Changed.